src/test/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicatorTest.java

```diff
@@ -481,4 +494,243 @@ public void testDoneReadingWithExceptionAndHasCompletionPacketLeftToRead() throw
+		advancedControllableProperties = extendedStatistic.getControllableProperties();
+
+		for (AdvancedControllableProperty property : advancedControllableProperties) {
+			if (property.getName().equalsIgnoreCase(Command.FOCUS.getName() + HASH + Command.FOCUS_MODE.getName())) {
+				Assert.assertTrue((int) property.getValue() == 1 || (int) property.getValue() == 0);
+			}
+		}
+	}
+
+	/**
+	 * Test AxisCommunicator#getMultipleStatistics success
+	 * Expect backlight status get data success
+	 */
+	@Test
+	public void testAverPTZCommunicatorBacklightStatus() {
+		extendedStatistic = (ExtendedStatistics) averPTZCommunicator.getMultipleStatistics().get(0);
+		advancedControllableProperties = extendedStatistic.getControllableProperties();
+
+		for (AdvancedControllableProperty property : advancedControllableProperties) {
+			if (property.getName().equalsIgnoreCase(Command.EXPOSURE.getName() + HASH + Command.BACKLIGHT.getName())) {
+				Assert.assertTrue((int) property.getValue() == 1 || (int) property.getValue() == 0);
+			}
+		}
+	}
+
+	/**
+	 * Test AxisCommunicator#getMultipleStatistics success
+	 * Expect backlight status get data success
+	 */
+	@Test
+	public void testAverPTZCommunicatorSlowPanTiltStatus() {
+		extendedStatistic = (ExtendedStatistics) averPTZCommunicator.getMultipleStatistics().get(0);
+		advancedControllableProperties = extendedStatistic.getControllableProperties();
+
+		for (AdvancedControllableProperty property : advancedControllableProperties) {
+			if (property.getName().equalsIgnoreCase(Command.PAN_TILT_DRIVE.getName() + HASH + Command.SLOW_PAN_TILT.getName())) {
+				Assert.assertTrue((int) property.getValue() == 1 || (int) property.getValue() == 0);
+			}
+		}
+	}
+
+	/**
+	 * Test AxisCommunicator#getMultipleStatistics success
+	 * Expect AE mode get data success
+	 */
+	@Test
+	public void testAverPTZCommunicatorAEMode() {
+		extendedStatistic = (ExtendedStatistics) averPTZCommunicator.getMultipleStatistics().get(0);
+		advancedControllableProperties = extendedStatistic.getControllableProperties();
+		stats = extendedStatistic.getStatistics();
+
+		String aeMode = "FullAuto", value;
+		List<String> shutterValues = new ArrayList<>(
+				Arrays.asList("1/32K", "1/16K", "1/8K", "1/4K", "1/2K", "1/1K", "1/480", "1/240", "1/120", "1/60", "1/30", "1/20", "1/10", "1/5", "1/2", "1/1"));
+		List<String> irisLevels = new ArrayList<>(
+				Arrays.asList("0", "F14", "F11", "F8.0", "F6.8", "F5.6", "F4.8", "F4.0", "F3.4", "F2.8", "F2.4", "F2.0", "F1.8", "F1.6"));
+
+		for (AdvancedControllableProperty property : advancedControllableProperties) {
+			if (property.getName().equalsIgnoreCase(Command.EXPOSURE.getName() + HASH + Command.AE_MODE.getName())) {
+				aeMode = (String) property.getValue();
+				Assert.assertTrue(aeMode.equals("FullAuto") || aeMode.equals("IrisPriority") || aeMode.equals("ShutterPriority") || aeMode.equals("Manual"));
+			}
+		}
+
+		switch (aeMode) {
```

Comment:
I know it's the way to handle all cases but this can limit us to identify the incorrect behavior based on the test case.
So I suggest to separate them to track which is incorrect easier

Reply:
yep thank you, I will do it